client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -1,13 +1,65 @@
 import json
 import math
 import os
-
+from typing import List
 from ayon_api import get_representation_by_id
 from ayon_maya.api import plugin
 from maya import cmds
 from maya.api import OpenMaya as om
 
 
+def convert_matrix_to_4x4_list(
+        value) -> List[List[float]]:
+    """Convert matrix or flat list to 4x4 matrix list
+    Example:
+        >>> convert_matrix_to_4x4_list(om.MMatrix())
+        [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]
+        >>> convert_matrix_to_4x4_list(
+        ... [1, 0, 0, 0,
+        ...  0, 1, 0, 0,
+        ...  0, 0, 1, 0,
+        ...  0, 0, 0, 1]
+        ... )
+        [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 1]]
+    """
+    result = []
+    value = list(value)
+    for i in range(0, len(value), 4):
+        result.append(list(value[i:i + 4]))
+    return result
+
+
+def convert_transformation_matrix(transform_mm: om.MMatrix, rotation: list) -> om.MMatrix:
+    """Convert matrix to list of transformation matrix for Unreal Engine import.
+
+    Args:
+        transform_mm (om.MMatrix): World Matrix for the asset
+        rotation (list): Rotations of the asset
```

Comment:
Why do we need to specify the rotation separately? The rotation should be in the `MMatrix` as well? Can't we just extract it from it? Why does it need to be a separate input argument?

Reply:
we need to readjust the rotation input from the cmds.xform(rotation=True), instead of using the one from the matrix, we are using the one from rotation. It's metathetically easier, if you have any other solution which is helpful for this case, feel free to suggest.

Reply:
Ah I see. You're using the raw rotation values from the node. I have a very odd feeling that this will break in many cases because rotation values of e.g. 360+ like 720 or 1080 which in essence are then also 0 rotation as a result may break in the same way in Unreal yet we're now not catching those?

Reply:
> Ah I see. You're using the raw rotation values from the node. I have a very odd feeling that this will break in many cases because rotation values of e.g. 360+ like 720 or 1080 which in essence are then also 0 rotation as a result may break in the same way in Unreal yet we're now not catching those?

Let add it as an issue and move discussion on that later. If this can be exported from maya, but it crashed in unreal. It can be done in either unreal or real. You can clamp the value in Unreal